### PR TITLE
added support for specifying json convenience object

### PR DESCRIPTION
**Pull request by blagerweij**
Allows you to conveniently use `json` field for multipart forms:

```
        http:
          url: https://httpbin.org/post
          method: POST
          formData:
            myfield:
              json:
                foo: bar
                bar: foo
```
The YAML will be converted to JSON, and the content-type set to 'application/json'.

### CODE REVIEW

File: src/steps/http.ts [hunk 1]
```diff
@@ -286,10 +291,6 @@ export default async function (
       const appendOptions = {} as FormData.AppendOptions
       if (typeof params.formData[field] === 'string') {
```
Comment (mishushakov) [inline src/steps/http.ts:292]: One thing: Can we also check for boolean and number? They should be converted automatically to string if you pass them to formData.append

**Review by mishushakov — COMMENTED** [review]
Overall looks good to me and I'd be happy to merge. Feel free to act on my comment.

File: src/steps/http.ts (hunk 1 above)
Comment (blagerweij) [reply src/steps/http.ts]: done, please check

**Comment by mishushakov** [comment]
Okay looks good to me now. Have you tested it?

**Comment by blagerweij** [comment]
Yes, I have also added a 'jsonfield' in the mulltipart.yaml test-file.

**Comment by mishushakov** [comment]
Do you actually use `HTTPRequestPartJson` type anywhere?

**Comment by blagerweij** [comment]
@mishushakov nice catch! I had initially added that type, but later decided to combine the json field in the existing HttpRequestPart type. I've updated the PR (removed the unused code); Also added a testcase for JSON arrays. I've tested it locally, and for me it works.

**Comment by mishushakov** [comment]
Very well done. One thing though: you can let the `!= null` part out, I don't think it makes any difference.

**Comment by blagerweij** [comment]
What happens if you use the following (valid) json:

```
          my_json_field:
              json: false
```

If I'm not wrong, then `if (requestPart.json != null)` works as expected, where `if (requestPart.json)` does not ?!?

**Comment by mishushakov** [comment]
Yes, the problem here will be if I want my json to be null. What you actually need is `if('json' in requestPart)`

**Comment by mishushakov** [comment]
Thank you very much! We should also fix the same problem in other places.

**Pull request merged**